CMakeLists.txt

```diff
@@ -1,21 +1,31 @@
 cmake_minimum_required(VERSION 3.2)
-project(fzf C)
+project(fzf LANGUAGES C)
 
 add_library(${PROJECT_NAME} SHARED "src/fzf.c")
 
+set(is-windows $<STREQUAL:${CMAKE_SYSTEM_NAME},Windows>)
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>)
+target_compile_options(${PROJECT_NAME}
+  PRIVATE
+    $<$<C_COMPILER_ID:MSVC>:/W4>
+    $<$<NOT:$<C_COMPILER_ID:MSVC>>:-Wall>)
+target_compile_definitions(${PROJECT_NAME}
+  PRIVATE
+    $<${is-windows}:_CRT_NONSTDC_NO_DEPRECATE>
+    $<${is-windows}:_CRT_SECURE_NO_DEPRECATE>
+    $<${is-windows}:_CRT_SECURE_NO_WARNINGS>)
```

Comment:
```suggestion
target_compile_definitions(${PROJECT_NAME}
  PRIVATE
    $<$<BOOL:${WIN32}>:_CRT_NONSTDC_NO_DEPRECATE>
    $<$<BOOL:${WIN32}>:_CRT_SECURE_NO_DEPRECATE>
    $<$<BOOL:${WIN32}>:_CRT_SECURE_NO_WARNINGS>)
```

Reply:
I forgot I can just use the `PLATFORM_ID` generator expression. This is what is being used now.